02-microarray/ortholog-mapping_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,371 @@
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(hcop)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+## Import and set up data
+
+Data downloaded from refine.bio include a metadata tab separated values (TSV) file and a data TSV file. 
+This chunk of code will read the both TSV files and add them as data frames to your environment.
+
+```{r}
+# Read in metadata TSV file
+metadata <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your metadata file
+  "metadata_GSE71270.tsv" # Replace with the name of your metadata file
+))
+
+# Read in data TSV file
+df <- readr::read_tsv(file.path(
+  data_dir, # Replace with path to your data file
+  "GSE71270.tsv" # Replace with the name of your data file
+)) %>%
+  # Tuck away the Gene ID column as rownames
+  tibble::column_to_rownames("Gene")
+```
+
+Let's ensure that the metadata and data are in the same sample order. 
+
+```{r}
+# Make the data in the order of the metadata
+df <- df %>%
+  dplyr::select(metadata$geo_accession)
+
+# Check if this is in the same order
+all.equal(colnames(df), metadata$geo_accession)
+
+# Bring back the "Gene" column in preparation for mapping
+df <- df %>%
+  tibble::rownames_to_column("Gene")
+```
+
+## Mapping zebrafish Ensembl gene IDs to human symbols
+
+refine.bio data uses Ensembl gene identifiers, which will be in the first column.
+
+```{r}
+# Let's take a look at the first 6 items in the `Gene` column of `df`
+head(df$Gene)
+```
+
+Enembl gene identifiers have different species-specific prefixes. 
+In zebrafish, Ensembl gene identifiers begin with `ENSDARG` (in human, `ENSG`, etc.).
+
+Now let's do the conversion!
+We'll use the `zebrafish` data from `hcop`; this is what `zebrafish` looks like.
+
+```{r}
+zebrafish
+```
+
+We're interested in the `human_symbol` and `zebrafish_ensembl` columns specifically.
+
+```{r}
+human_zebrafish_key <- zebrafish %>%
+  # we'll want to subset zebrafish to only the columns we're interested in
+  dplyr::select(zebrafish_ensembl, human_symbol)
+
+# Now let's join the expression data from `df` for the subset of genes in `df`
+human_zebrafish_mapped_df <- df %>%
+  dplyr::left_join(human_zebrafish_key, by = c("Gene" = "zebrafish_ensembl"))
+```
+
+Here's what the new data frame looks like:
+
+```{r}
+head(human_zebrafish_mapped_df, n = 25)
+```
+
+Let's get a summary of the gene symbols returned in the `human_symbol` column of our mapped data frame.
+
+```{r}
+# We can use the `count()` function to get a count of the distribution of values in the `human_symbol` column
+human_zebrafish_mapped_df %>%
+  dplyr::group_by(human_symbol) %>%
+  dplyr::count() %>%
+  # Sort by highest `n` which would be the human gene symbol with the most mapped zebrafish Ensembl IDs
+  dplyr::arrange(desc(n))
+```
+
+Looks like we have mapped symbols!
+We also have 415 NAs, which is okay, because we do not expect everything to map across species.
```

Comment:
```suggestion
We also have 415 NAs, which is okay, because we do not expect everything to map across species.
```
This finding, although good to have, is not clear from the output shown here, can add a small chunk that shows how to get this number?